src/d3a/models/powerflow/__init__.py

```diff
@@ -0,0 +1,123 @@
+
+
+class PowerFlowBase(ABC):
+    def __init__(self, root_area, root_voltage):
+        """
+        :param root_area: contains tree structured hierarchical energy grid
+        :param root_voltage: contains the voltage of the top most hierarchy
+        """
+        self.root_voltage = root_voltage
+        self._grid_forming(root_area)
+
+    def _grid_forming(self, area):
+        """
+        :param area: takes area object and convert it to specific energy device
+        :return: electrical network compliant connected grid
+        """
+        if area.strategy is None:
+            area.bus = self.create_bus(area, self.root_voltage)
+            if area.parent is not None:
+                area.line = self.add_line(area.parent, area)
+        elif isinstance(area.strategy, InfiniteBusStrategy):
+            area.ext_grid = self.add_external_grid(area)
+        elif isinstance(area.strategy, LoadHoursStrategy):
+            area.load_device = self.add_load_device(area, area.strategy.avg_power_W)
+        elif isinstance(area.strategy, StorageStrategy):
+            area.ess_device = self.add_storage_device(area)
+        elif isinstance(area.strategy, PVStrategy):
+            area.pv_device = \
+                self.add_generation_device(area, area.strategy.max_panel_power_W * 1e-3)
+        elif isinstance(area.strategy, FinitePowerPlant):
+            area.power_plant = \
+                self.add_generation_device(area, area.strategy.max_available_power_kW)
+        for child in area.children:
+            self._grid_forming(child)
+
+    @abstractmethod
+    def create_bus(self, area, voltage):
+        """
+        :param area: area node where bus bar needs to be added
+        :param voltage: voltage level of bus bar
+        :return: add a bus-bar to the specified area node
+        """
+        pass
+
+    @abstractmethod
+    def add_external_grid(self, area):
+        """
+        :param area: area node where external grid element needs to be added
+        :return: an external grid device like MarketMaker
+        """
+        pass
+
+    @abstractmethod
+    def add_load_device(self, area, avg_power_w):
+        """
+        :param area: Takes area that is using Load as a strategy
+        :param avg_power_w: Peak power of the load device
+        :return: add a load device to the specified area node
+        """
+        pass
+
+    @abstractmethod
+    def add_generation_device(self, area, peak_power_kw):
+        """
+        :param area: Takes area that is using generating strategy
+        :param peak_power_kw: Takes into account power rating of the generating device
+        :return: add a generating device to the specified area node
+        """
+        pass
+
+    @abstractmethod
+    def add_storage_device(self, area):
+        """
+        :param area: Takes area that is using Storage as a strategy
+        :return: add an Storage device to the specified area node
```

Comment:
Sorry to nitpick here. Can you please put `Add a storage device to the specified area node` (there is also a typo) in the first line of the docstring and either remove the return line completely or describe what is returned (in my opinion not feasible).

Reply:
BTW: I mean this for all the docstrings in this class, sorry to write this so late.